src/pg/statement.cr

```diff
@@ -8,37 +8,79 @@ class PG::Statement < ::DB::Statement
   end
 
   protected def perform_query(args : Enumerable) : ResultSet
-    params = args.map { |arg| PQ::Param.encode(arg) }
-    conn = self.conn
-    conn.send_parse_message(command)
-    conn.send_bind_message params
-    conn.send_describe_portal_message
-    conn.send_execute_message
-    conn.send_sync_message
-    conn.expect_frame PQ::Frame::ParseComplete
-    conn.expect_frame PQ::Frame::BindComplete
-    frame = conn.read
-    case frame
-    when PQ::Frame::RowDescription
-      fields = frame.fields
-    when PQ::Frame::NoData
-      fields = nil
-    else
-      raise "expected RowDescription or NoData, got #{frame}"
+    retry do
+      begin
+        params = args.map { |arg| PQ::Param.encode(arg) }
```

Comment:
Can this be done outside the retry block?

Reply:
Why can't use the DB pool retry where the connection is dispatched rather than in the statement? https://github.com/crystal-lang/crystal-db/blob/master/src/db/pool.cr#L165-L191

Reply:
we can also do that, that's probably better